mne/io/pick.py

```diff
@@ -784,24 +788,163 @@ def _pick_aux_channels(info, exclude='bads'):
+    #
+    if isinstance(picks, slice):
+        picks = np.arange(n_chan)[picks]
+
+    #
+    # -> ndarray of int (and make a copy)
+    #
+    picks = np.atleast_1d(picks)  # this works even for picks == 'something'
+    picks = np.array([], int) if picks.size == 0 else picks
+    if picks.ndim != 1:
+        raise ValueError('picks must be 1D, got %sD' % (picks.ndim,))
+    if picks.dtype.char in ('S', 'U'):
+        picks = _picks_str_to_idx(info, picks, exclude, with_ref_meg)
+    if picks.dtype != np.int:
+        raise TypeError('picks must be a list of int or list of str, got '
+                        'a data type of %s' % (picks.dtype,))
+
+    #
+    # ensure we have (optionally non-empty) ndarray of valid int
+    #
+    if len(picks) == 0 and not allow_empty:
+        raise ValueError('No appropriate channels found for the given picks '
+                         '(%r)' % (orig_picks,))
+    if (picks < -n_chan).any():
+        raise ValueError('All picks must be >= %d, got %r'
+                         % (-n_chan, orig_picks))
+    if (picks >= n_chan).any():
+        raise ValueError('All picks must be < %d, got %r'
+                         % (n_chan, orig_picks))
+    picks %= n_chan  # ensure positive
+    return picks
+
+
+def _picks_str_to_idx(info, picks, exclude, with_ref_meg):
+    """Turn a list of str into ndarray of int."""
+    # special case for _picks_to_idx w/no info: shouldn't really happen
+    if isinstance(info, int):
+        raise ValueError('picks as str can only be used when measurement '
+                         'info is available')
+
+    #
+    # first: check our special cases
+    #
+
+    picks_generic = list()
+    if len(picks) == 1:
+        if picks[0] == 'all':
+            use_exclude = info['bads'] if exclude == 'bads' else exclude
+            picks_generic = pick_channels(info['ch_names'], info['ch_names'],
+                                          exclude=use_exclude)
+        elif picks[0] == 'data':
+            picks_generic = _pick_data_channels(info, exclude=exclude,
+                                                with_ref_meg=with_ref_meg)
+        elif picks[0] == 'data_or_ica':
+            picks_generic = _pick_data_or_ica(info, exclude=exclude)
+
+    #
+    # second: match all to channel names
+    #
+
+    bad_name = None
+    picks_name = list()
+    for pick in picks:
+        try:
+            picks_name.append(info['ch_names'].index(pick))
+        except ValueError:
+            bad_name = pick
+            break
+
+    #
+    # third: match all to types
+    #
+    bad_type = None
+    picks_type = list()
+    kwargs = dict(meg=False)
+    meg, fnirs = set(), set()
+    for pick in picks:
+        if pick in _PICK_TYPES_KEYS:
+            kwargs[pick] = True
+        elif pick in _MEG_CH_TYPES_SPLIT:
+            meg |= {pick}
+        elif pick in _FNIRS_CH_TYPES_SPLIT:
+            fnirs |= {pick}
+        else:
+            bad_type = pick
+            break
+    else:
```

Comment:
else seems to be at the same level as the for loop above

Reply:
Yes that's intentional, it means `if did not break`